src/en/authors-charm-leadership.md

```diff
@@ -0,0 +1,187 @@
+    should arrive soon).
+
+### ...share varying configuration among units?
+
+You should be able to use the exact same constructs as above, in the same way;
+you just might want to call `set_shared_settings` in a few more places. If you
+need additional synchronisation, you can use a peer relation to communicate
+minions' acknowledgements back to the leader.
+
+!!! Note: peer relation membership is not guaranteed to match current reality
+at any given time. To be resilient in the face of your service scaling at the
+same time as you (say) rebalance your service, your leader code will need to
+use the output of `status-get --service` to determine up-to-date membership,
+and wait for the set of acked units in the peer relation to match that list.
+
+### ...guarantee that a long-lived process runs on just one unit at once?
+
+The `hacluster` charm used in our OpenStack deploys will set up corosync and
+pacemaker, and may well be relevant to your needs; if that's not a good fit,
+read on.
+
+Juju's leadership concept is, by choice, relatively fine-grained, to ensure
+timely handover of agent-level responsibilities. That's why `is-leader` success
+guarantees only 30s of leadership; but it's no fun running a separate watchdog
+process to `juju-run is-leader` every 30s and kill your process when that stops
+working (apart from anything else, your juju-run could be blocked by other
+operations, so you can't guarantee a run every 30s anyway).
+
+And we don't plan to allow coarser-grained leadership requests. This is because
+if one unit could declare itself leader for a day (or even an hour) a failed
+leader will leave other parts of juju blocked for that length of time, and we're
+not willing to take on that cost; the 30-60s handover delay is bad enough
+already.
+
+So, you'd basically have to implement your own protocol on top of the available
+primitives. As a charm author, this is unlikely to be the best use of your time
+-- extending the `hacluster` charm to cover your use case is likely to be more
+efficient.
+
+### ...render it likely that a long-lived process runs on one unit at a time?
+
+!!! Note: this approach is not reliable. It may be good enough for some
+workloads, but don't use it unless you understand the forces in play and the
+worst possible consequences for your users...
+
+If you start your long-lived process in `leader-elected`, and stop it in
+`leader-settings-changed`, this will *usually* do what you want, but is
+vulnerable to a number of failure modes -- both because hook execution may be
+blocked until after the leadership lease has actually expired, *and* because
+total juju failure could also cause the hook not to run (but leave the workload
+untouched).
+
+In the future, we may implement a `leader-deposed` hook, that can run with
+*stronger* timeliness guarantees; but even if we do, it's a fundamentally
+unreliable approach. Seriously, if you possibly can, go with the charmed-up
```

Comment:
We've referenced hacluster a few times... but what exactly would we be re-using here? Seems vague to offer this without some direction as to the code samples they should be looking at, and then explaining why they are superior. (i also get that you've warned against this method above... but clarity)